public/app/core/services/backend_srv.ts

```diff
@@ -334,15 +324,33 @@ export class BackendSrv implements BackendService {
       message = err.message;
     }
 
+    // Validation
+    if (err.status === 422) {
+      description = err.data.message;
+      message = 'Validation failed';
+    }
+
+    if (config.showErrorAlert === false) {
+      return;
+    }
+
     if (message.length > 80) {
       description = message;
       message = 'Error';
     }
 
-    // Validation
-    if (err.status === 422) {
-      description = err.data.message;
-      message = 'Validation failed';
+    if (!err.statusText) {
+      err.statusText = 'error';
+    }
+
+    if (message.length > 80) {
+      description = message;
+      message = 'Error';
+    }
```

Comment:
I would do any changes like these in a separate, more atomic PR (it's easier to reason about them separately and probably more stakeholders will be involved).

Although a change like this (changing the error message and moving the info to the description) can make sense, it can affect different parts of the app or even plugins - the error emitted through the event bus, and if a certain part of the app was only using the error message until now, then after this change it would lose information on the UI.